synapse/lib/agenda.py

```diff
@@ -584,22 +584,19 @@ def list(self):
                 if appt.isrunning:
                     logger.warning(
                         'Appointment %s is still running from previous time when scheduled to run.  Skipping.',
-                        s_common.ehex(appt.iden))
+                        appt.iden)
                 else:
                     await self._execute(appt)
 
     async def _execute(self, appt):
         '''
         Fire off the task to make the storm query
         '''
-        if appt.username is None or self.core.auth is None:
-            user = None
-        else:
-            user = self.core.auth.getUserByName(appt.username)
-            if user is None:
-                logger.warning('Unknown username %s in stored appointment', appt.username)
-                return
-        await self.core.boss.execute(self._runJob(user, appt), f'Agenda {s_common.ehex(appt.iden)}', user)
+        user = self.core.auth.user(appt.useriden)
+        if user is None:
+            logger.warning('Unknown user %s in stored appointment', appt.useriden)
```

Comment:
should we mark the appointment here as failed?

Reply:
ok